src/main/java/io/nats/client/StreamContext.java

```diff
@@ -13,24 +13,185 @@
+     * @return stream information
+     * @throws IOException covers various communication issues with the NATS
+     *         server such as timeout or interruption
+     * @throws JetStreamApiException the request had an error related to the data,
+     *         most likely the stream has been removed since the context was created.
+     */
+    StreamInfo getStreamInfo() throws IOException, JetStreamApiException;
+
+    /**
+     * Gets information about the stream for this context.
+     * @param options the stream info options. If null, request will not return any optional data.
+     * @return stream information
+     * @throws IOException covers various communication issues with the NATS
+     *         server such as timeout or interruption
+     * @throws JetStreamApiException the request had an error related to the data,
+     *         most likely the stream has been removed since the context was created.
+     */
+    StreamInfo getStreamInfo(StreamInfoOptions options) throws IOException, JetStreamApiException;
+
+    /**
+     * Purge stream messages
+     * @return PurgeResponse the purge response
+     * @throws IOException covers various communication issues with the NATS
+     *         server such as timeout or interruption
+     * @throws JetStreamApiException the request had an error related to the data
+     */
+    PurgeResponse purge() throws IOException, JetStreamApiException;
+
+    /**
+     * Purge messages for a specific subject
+     * @param options the purge options
+     * @return PurgeResponse the purge response
+     * @throws IOException covers various communication issues with the NATS
+     *         server such as timeout or interruption
+     * @throws JetStreamApiException the request had an error related to the data
+     */
+    PurgeResponse purge(PurgeOptions options) throws IOException, JetStreamApiException;
+
+    /**
+     * Create a consumer context for on the context's stream and specific named consumer.
+     * Verifies that the consumer exists.
+     * @param consumerName the name of the consumer
+     * @return a ConsumerContext object
+     * @throws IOException covers various communication issues with the NATS
+     *         server such as timeout or interruption
+     * @throws JetStreamApiException the request had an error related to the data
+     */
+    ConsumerContext consumerContext(String consumerName) throws IOException, JetStreamApiException;
+
+    /**
+     * Management function to creates a consumer on this stream.
+     * @param config the consumer configuration to use.
+     * @return consumer information.
+     * @throws IOException covers various communication issues with the NATS
+     *         server such as timeout or interruption
+     * @throws JetStreamApiException the request had an error related to the data
+     */
+    ConsumerContext addConsumer(ConsumerConfiguration config) throws IOException, JetStreamApiException;
```

Comment:
Just FYI - not sure which ack policy is default right now, but we recently agreed that at least for the js simplification part (to avoid breaking changes in the old implementation) we should default to `AckExplicit`). I mention this because go client uses `AckNone` as default in old implementation and I just changed it for simplified API.

Reply:
In Java/.NET, when you are creating a ConsumerConfiguration through code, Explicit is the default but of course can be set/changed by the developer. I'm not sure if I understand the comment, this method is accepting the ConsumerConfiguration that the user has supplied, so I can only assume that it's configured the way they want it.

Reply:
Sure, I just put the comment here because I was curious about the default (having in mind the recent issue in go client) - if it's explicit, that's correct